### PR TITLE
Lambdatest integration

### PR DESCRIPTION
**Pull request by mukesh-lt**
### Description
This PR adds capability to run karma tests on LambdaTest platform

### Type of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would change existing functionality)

### Checklist

- [ ] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)
- [ ] My code follows the code style of the project _(using `npm run lint`)_
- [ ] My change introduces changes to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

#### Live previews

- <https://deploy-preview-{your_pr_number}--twbs-bootstrap.netlify.app/>

### Related issues

@XhmikosR 
### Steps to run tests on LambdaTest

- in the root folder do npm install and export below variables
    - `export LT_USERNAME=username`
    - `export LT_ACCESS_KEY=accessKey`
    - `export LT_TUNNEL_NAME=tunnelName` (Optional)
 - Run tests using command `npm run js-test-ltcloud`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -72,6 +72,7 @@
     "js-test-integration-bundle": "rollup --config js/tests/integration/rollup.bundle.js",
     "js-test-integration-modularity": "rollup --config js/tests/integration/rollup.bundle-modularity.js",
     "js-test-cloud": "cross-env BROWSERSTACK=true npm run js-test-karma",
+    "js-test-ltcloud": "cross-env LAMBDATEST=true & node js/tests/lt-local & sleep 5  &  npm run js-test-karma",
```
Comment (XhmikosR) [inline package.json:75]: This is is using unix specific features which we generally avoid. Like `&` and `sleep`. For `&` we use `npm-run-all --parallel`. For sleep let's leave it for the time being.

File: package.json [hunk 2]
```diff
@@ -136,7 +138,9 @@
     "karma-firefox-launcher": "^2.1.2",
     "karma-jasmine": "^5.1.0",
     "karma-jasmine-html-reporter": "^2.0.0",
+    "karma-mocha-reporter": "^0.3.1",
```
Comment (XhmikosR) [inline package.json:141]: This is unused

File: js/tests/lt-local.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+const tunnelInstance = new lambdaTunnel();
+const tunnelArguments = {
+    user: process.env.LT_USERNAME,
+    key: process.env.LT_ACCESS_KEY,
+    tunnelName: process.env.LT_TUNNEL_NAME || 'jasmine',
+    logFile: "local.log"
+};
+
+if (process.env.LAMBDATEST == 'true')
+    tunnelInstance
```
Comment (XhmikosR) [inline js/tests/lt-local.js:11]: could you switch these to async/await?

Comment (mukesh-lt) [reply js/tests/lt-local.js]: @XhmikosR sure, addressed comments

**Comment by XhmikosR** [comment]
I still can't edit your files :/ You should see a checkbox on the right here in the PR, please enable it:

image

**Comment by vipulkv** [comment]
hi @XhmikosR - i have added you as maintainer 
image

**Comment by XhmikosR** [comment]
Thanks but it shouldn't be needed... Out of curiosity, don't you see the checkbox here in the PR?

**Comment by vipulkv** [comment]
Nope. I don't see option in the PR.

**Comment by XhmikosR** [comment]
@mukesh-lt could you please add @GeoSot to your repo too?

**Comment by vipulkv** [comment]
Done
image

**Comment by XhmikosR** [comment]
@vipulkv thank you! We merged it in a branch in our repo so that we can test it and we'll let you know how it goes.

**Comment by XhmikosR** [comment]
OK, it seems it fails with:

```
Error checkPath -----> Error: ENOENT: no such file or directory, access '/home/runner/work/bootstrap/bootstrap/.lambdatest/v3'
```

**Comment by XhmikosR** [comment]
Then we got:

```
Downloading latest binary
Extracting binary
Starting tunnel
Tunnel successfully initiated. You can start testing now
```

But we see nothing on LambdaTest.

**Comment by mukesh-lt** [comment]
> Then we got:
> 
> ```
> Downloading latest binary
> Extracting binary
> Starting tunnel
> Tunnel successfully initiated. You can start testing now
> ```
> 
> But we see nothing on LambdaTest.

@XhmikosR there are two commands being running, this shows that first command worked and you should 1 active tunnel in LambdaTest account, within few seconds we should see the tests start running

**Comment by XhmikosR** [comment]
@mukesh-lt thanks! It seems some tests are now failing due to timeout https://github.com/twbs/bootstrap/actions/runs/4543010434/jobs/8007150779

**Comment by mukesh-lt** [comment]
> @mukesh-lt thanks! It seems some tests are now failing due to timeout https://github.com/twbs/bootstrap/actions/runs/4543010434/jobs/8007150779

@XhmikosR it seems all the mobile tests are failing, could you plz confirm you have the `Mobile browser on Real Device` in your subscription plan, if you do not wish to run tests on real device then you can comment them in browsers.js file

**Comment by XhmikosR** [comment]
Thanks I sent an email and I'm waiting to hear back!

**Comment by vipulkv** [comment]
@XhmikosR We have enabled Web Automation on Real Devices for your subscription. Can you try running the test on Real Device now and let us know if you face any issues.

**Comment by SparshKesari** [comment]
Hey @XhmikosR,
Can you please rerun the tests? It will help diagnose the issue.

**Comment by XhmikosR** [comment]
The fourth time I restarted the job, tests passed. I'm going to restart CI again to see if it happens again.

**Comment by SparshKesari** [comment]
Hey @XhmikosR,
We are looking into this.

**Pull request merged**